software/bwamem2/index/main.nf

```diff
@@ -0,0 +1,35 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BWAMEM2_INDEX {
+    tag "$fasta"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:'') }
+
+    conda (params.enable_conda ? "bioconda::bwa-mem2=2.1=he513fc3_0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bwa-mem2:2.1--he513fc3_0"
+    } else {
+        container "quay.io/biocontainers/bwa-mem2:2.1--he513fc3_0"
+    }
+
+    input:
+    path fasta
+
+    output:
+    path "bwa"          , emit: index
```

Comment:
```suggestion
    path "bwamem2"          , emit: index
```

Reply:
I wonder if its worth making this distinction here just in case you need to have both `bwa` (original) and `bwamem2` indices together in the same folder for example. Would save renaming.

Reply:
I don't really see people using both bwa and bwamem2 in the same pipeline at once, but you have a valid point